src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs.rs

```diff
@@ -0,0 +1,337 @@
+                    }).as_ref().cloned();
+                    if let Some((method_pointer,expression_id)) = info {
+                        self.breadcrumb_push.emit((method_pointer,expression_id));
+                    }
+                }
+            },
+            cmp::Ordering::Equal => ()
+        }
+        info!(self.logger,"Selecting breadcrumb #{index}");
+    }
+
+    fn push_breadcrumb(&self, method_pointer:&Rc<MethodPointer>, expression_id:&uuid::Uuid) {
+        let current_index = self.current_index.get();
+        let next_index    = current_index + 1;
+
+        let breadcrumb_exists =
+            self.breadcrumbs.borrow_mut().get(current_index).map(|breadcrumb| {
+                breadcrumb.info.expression_id
+            }).filter(|other_expression_id| other_expression_id == expression_id).is_some();
+
+        if breadcrumb_exists {
+            debug!(self.logger, "Entering an existing {method_pointer.name} breadcrumb.");
+            //TODO[dg]: Highlight breadcrumb.
+        } else {
+            debug!(self.logger, "Creating a new {method_pointer.name} breadcrumb.");
+            self.remove_breadcrumbs_history();
+            let breadcrumb       = Breadcrumb::new(&self.scene,method_pointer,expression_id);
+            let network          = &breadcrumb.frp.network;
+            let breadcrumb_index = next_index;
+            let model            = self.clone_ref();
+
+
+            // === User Interaction ===
+
+            frp::extend! { network
+                eval_ breadcrumb.frp.outputs.selected(model.select_breadcrumb(breadcrumb_index));
+            }
+
+
+            // === GUI Update ===
+
+            frp::extend! { network
+                eval self.project_name.frp.outputs.width((width) {
+                        model.set_project_name_width(*width)
+                });
+            }
+
+            info!(self.logger, "Pushing {breadcrumb.info.method_pointer.name} breadcrumb.");
+            breadcrumb.set_position(Vector3::new(self.width(), 0.0, 0.0));
+            breadcrumb.frp.fade_in.emit(());
+            self.breadcrumbs_container.add_child(&breadcrumb);
+            self.breadcrumbs.borrow_mut().push(breadcrumb);
+        }
+        self.current_index.set(next_index);
+        self.update_selection();
+    }
+
+    fn pop_breadcrumb(&self) {
+        if self.current_index.get() > 0 {
+            info!(self.logger, "Popping breadcrumb view.");
+            self.current_index.set(self.current_index.get() - 1);
+        }
+        self.update_selection();
+    }
+
+    fn update_selection(&self) {
+        let current_index = self.current_index.get() - 1;
+        for (index,breadcrumb) in self.breadcrumbs.borrow_mut().iter().enumerate() {
+            if index == current_index {
+                breadcrumb.frp.select.emit(());
+            } else {
+                breadcrumb.frp.deselect.emit(());
+            }
+        }
```

Comment:
Probably it will be fixed according to the Wojciech's comment, but as reminder: *Never* emit events or call callbacks, or even call non-trivial functions while Refcell is borrowed. You never know what will be attached to the event/callback in future.